### DISCUSSION TITLE
5+4 Critical gaps / Woo FSE pure players

### DISCUSSION THREAD

**Discussion by Marc-pi** [woocommerce/woocommerce#44945] [WooCommerce, Blocks, and Site Editing]
I'm adding this here to hightlight the missing features for anyone willing to play with Woo + Gutenberg + Block Theme
Many very nice structual things are being worked on under the hoods (new product editor, Product Collection, CYS, HPOS, Gallery, nice patterns, checkout/cart as default, etc.)  but some points are still missing to facilitate our projects under the new paradigm (block editor) aka easy store editing
Here i'm focussing on top needs. 

**The goal of this post**
In an ocean of declared issues : 
- Help the Woo Blocks team to adjust the roadmap (have a better balance between essential needs and new innovative features)
- Help us check / recheck the current availability of critical needs before starting a new project
- Facilitate onboarding for anyone willing to play with WooCommerce + Gutenberg

> Block editing is fun, let's fill in the critical gaps quickly :-)

**Current critical missing features**

- [ ] **Finalize the blockification of the Single product template** : some nested blocks are too compact / offer poor flexibility
  - [ ] Add to cart with options block > button style > cannot be altered https://github.com/woocommerce/woocommerce-blocks/issues/11005
   - [ ] Product detail block is not flexible, we cannot alter its content https://github.com/woocommerce/woocommerce-blocks/issues/10366
   - [ ] attribute block/category block settings : add flexibility within the template https://github.com/woocommerce/woocommerce-blocks/issues/10365
- [ ] **Grouped Products Page** : https://github.com/woocommerce/woocommerce-blocks/issues/10363
no way to alter the way child rows are displayed (those are encapsulated within the compact Add to Cart with Options block
no dedicated template neither, but could be an optional way based on the Single Product Template
the classic need : add the SKU for each child element, an Add to cart button per child line, change the quantity selector style (-/+)
- [ ] **Add to cart with option block** > missing ... options !
  - [ ] Quantity selector type : choose between the compact selector and an extended one (-/+) https://github.com/woocommerce/woocommerce-blocks/issues/10225
  - [ ] Add to cart with options blocks > button style > see above in Single Product template (https://github.com/woocommerce/woocommerce-blocks/issues/11005)
  - [ ] see above the Grouped Product issue : this block encapsulate the product lines, no way to alter their design or to alter the lines with new content blocks
  - [ ] Advanced button styling > it's a missing WP feature (targeted for 6.5) > a:hover / interactive states, shadows
- [ ] **Collection Filters Block** 
     - [ ] dynamic nested filters https://github.com/woocommerce/woocommerce-blocks/issues/10985#issuecomment-1722872497
     - [ ] dropdown UX https://github.com/woocommerce/woocommerce-blocks/issues/11021
- [ ] **Product Collection > Product Image / Image Gallery > Out of stock badge setting**
      - https://github.com/woocommerce/woocommerce-blocks/issues/10406
      - https://github.com/woocommerce/woocommerce-blocks/issues/6688

**Critical Bugs**
- [x] category hardcoded / not translatable https://github.com/woocommerce/woocommerce-blocks/issues/11758
- [ ] product summary/ WP excerpt issue with 6.3 https://github.com/woocommerce/woocommerce/issues/39934
- [ ] extra padding in FSE template canva https://github.com/woocommerce/woocommerce-blocks/issues/10671
- [ ] quantity selectors not reachable or invisible on mobile https://github.com/woocommerce/woocommerce-blocks/issues/11037
- [ ] Shop page title hardcoded https://github.com/woocommerce/woocommerce-blocks/issues/10336

-----

**Other quick wins**
   - [ ] sort by default Reviews D with no custom filter https://github.com/woocommerce/woocommerce-blocks/issues/10647
   - [ ] shop page inconsistencies https://github.com/woocommerce/woocommerce-blocks/issues/10765
   - [ ] Product collection > product type option https://github.com/woocommerce/woocommerce-blocks/issues/10983
   - [ ] External / Affiliate product type : i did not check...
   - [ ] Set sale prices for variable product / bulk issue : https://github.com/woocommerce/woocommerce/issues/36802

**Comment by ralucaStan** [comment]
Hey @Marc-pi, thank you so much for summarizing your thoughts on the product gaps.

Commenting on behalf of the checkout blocks team, what exactly do you mean by  `Alter button style directly from block's settings`.

**Comment by Marc-pi** [comment]
@ralucaStan good point, i've adjusted the phrasing. It's a Gutenberg missing feature (targeted in 6.5)

**Comment by Marc-pi** [comment]
@ralucaStan gotcha, i got it : the style options are not available in Single Product Template for instance see https://github.com/woocommerce/woocommerce-blocks/issues/11005 // i've added it in the "complete the blockification part" and "Add to cart with options" part. it's related, but a major gap !
